plugins/plugin-hosted/codenvy-machine-hosted/src/main/java/com/codenvy/swarm/client/AwsInitialAuthConfig.java

```diff
@@ -0,0 +1,101 @@
+/*
+ *  [2012] - [2016] Codenvy, S.A.
+ *  All Rights Reserved.
+ *
+ * NOTICE:  All information contained herein is, and remains
+ * the property of Codenvy S.A. and its suppliers,
+ * if any.  The intellectual and technical concepts contained
+ * herein are proprietary to Codenvy S.A.
+ * and its suppliers and may be covered by U.S. and Foreign Patents,
+ * patents in process, and are protected by trade secret or copyright law.
+ * Dissemination of this information or reproduction of this material
+ * is strictly forbidden unless prior written permission is obtained
+ * from Codenvy S.A..
+ */
+package com.codenvy.swarm.client;
+
+import com.google.inject.Singleton;
+import com.google.inject.name.Named;
+
+import org.eclipse.che.commons.annotation.Nullable;
+
+import javax.inject.Inject;
+
+import static com.google.common.base.Strings.isNullOrEmpty;
+
+/**
+ * Keeps auth configurations for AWS Elastic Container Registry.
+ * Credential might be configured in .properties files.
+ *
+ * @author Mykola Morhun
+ */
+@Singleton
+public class AwsInitialAuthConfig {
+
+    private final String awsAccountId;
+    private final String region;
+    private final String accessKeyId;
+    private final String secretAccessKey;
+    private final String ecr;
+
+    @Inject
+    public AwsInitialAuthConfig(@Nullable @Named("aws.id") String awsId,
```

Comment:
I disagree. The main task of `AuthResolver` is match credentials to requested registry. It shouldn't care about configs reading, validation of it, etc.